### ISSUE TITLE
null compliance

### ISSUE THREAD

**Issue by jammasterj89** [mozilla/ssh_scan#379]
I'm running the latest docker ssh_scan as:

`docker run -it mozilla/ssh_scan /app/bin/ssh_scan -t hostname`

But I am receiving null as the ssh_version and for all of the compliance section (I've cropped the middle of the output):

```json
{
    "ssh_scan_version": "0.0.21",
    "ip": "xxxxxx",
    "hostname": "xxxxx",
    "port": 22,
    "server_banner": "SSH-2.0-OpenSSH_7.2p2 Ubuntu-4ubuntu2.1",
    "ssh_version": null,
    "os": "ubuntu",
    "os_cpe": "o:canonical:ubuntu:16.04",
    "ssh_lib": "openssh",
    "ssh_lib_cpe": "a:openssh:openssh:7.2p2",
    "key_algorithms": [
      "curve25519-sha256@libssh.org",
      "ecdh-sha2-nistp256",
      "ecdh-sha2-nistp384",
      "ecdh-sha2-nistp521",
      "diffie-hellman-group-exchange-sha256",
      "diffie-hellman-group14-sha1"
    ],
    "compliance": {
      "policy": null,
      "compliant": null,
      "recommendations": null,
      "references": null,
      "grade": null
    },
    "start_time": "2017-06-07 14:48:26 +0000",
    "end_time": "2017-06-07 14:48:26 +0000",
    "scan_duration_seconds": 0.333129422
  }
]
```

Both client and server are Ubuntu 16.04 with OpenSSH version: 

`OpenSSH_7.2p2 Ubuntu-4ubuntu2.1, OpenSSL 1.0.2g  1 Mar 2016`

Not sure where I am going wrong? There is no client config file under ~/.ssh/config (for all users), instead the client and server config both exist under /etc/ssh/ssh_config and sshd_config respectively. Could this be the reason for the null results?

Thank you!

**Comment by claudijd** [comment]
@jammasterj89 this is new behavior to me.  Also, truthfully, I run mostly the source checkout and/or gem install, and I don't have any integration tests around the docker image just yet.  I'll pull down the latest and see if I can replicate the issue.  Thanks for the report!

**Comment by claudijd** [comment]
I've verified this behavior on the latest version of the docker image, I'll try to see if I can replicate outside of docker and find out where the break down is.

**Comment by claudijd** [comment]
source installs and gem installs on master are not affected...

**From source:**
ruby bin/ssh_scan -t ssh.mozilla.com (this works fine)

**From gem:**
ssh_scan -t ssh.mozilla.com (this works fine)

**Comment by claudijd** [comment]
Shelling into the docker container proves a similar result...

$ docker run -it mozilla/ssh_scan /bin/bash
root@c6b63e2c4c77:/app# ./bin/ssh_scan -t ssh.mozilla.com

I'll trace the code in the container and see what's happening here.

**Comment by claudijd** [comment]
Ok, I think I figured out what the problem is...

I believe that somehow the travis runs for the docker publish is not being restricted to the master branch and because of this the docker content is based on this branch...

https://github.com/mozilla/ssh_scan/pull/378

I believe this is the case because when doing some debugging, I found the result object is a SSHScan::Result when it should be a Hash based on the code in master.

This explains why the source and gem installs above are fine, but the docker image is hosed.

**Comment by claudijd** [comment]
Ok, I believe I just landed a fix for this, but it shouldn't be live until the master branch build is triggered and the latest master rev is build/updated against the docker hub.  Once that is done, I will re-pull the docker image and then validate this is truly fixed for you.

**Comment by claudijd** [comment]
@jammasterj89 you should be able to pull and retry your scan now and see if that fixes you up.  Please let me know if you run into any other problems and I'll do my best to provide a timely fix.

**Comment by jammasterj89** [comment]
Thank you @claudijd for the super quick fix, I've just pulled the latest and everything looks good!

**Comment by jammasterj89** [comment]
Hi @claudijd, I've just noticed the grading is no longer appearing in the results after this latest docker pull, is that something you are seeing also?

**Comment by jammasterj89** [comment]
The grading was fixed in v0.0.25, thank you.
